docs/src/writing_good_rules.md

```diff
@@ -63,6 +63,39 @@ Examples being:
 - There is only one derivative being returned, so from the fact that the user called
   `frule`/`rrule` they clearly will want to use that one.
 
+## Use `ProjectTo` appropriately
+
+Rules with abstractly-typed arguments may return incorrect answers when called with certain concrete types.
+A classic example is the matrix-matrix multiplication rule
+```
+function rrule(::typeof(*), A::AbstractMatrix, B::AbstractMatrix)
+    function times_pullback(ȳ)
+        dA = ȳ * B'
+        dB = A' * ȳ
+        return NoTangent(), dA, dB
+    end
+    return A * B, times_pullback 
+end
+```
+When computing `*(A, B)`, where `A isa Diagonal` and `B isa Matrix`, the output will be a `Matrix`.
+As a result, `ȳ` in the pullback will be a `Matrix`, and consequently `dA` for a `A isa Diagonal` will be a `Matrix`, which is wrong.
```

Comment:
```suggestion
As a result, `ȳ` in the pullback will be a `Matrix`, and consequently `dA` for a `A isa Diagonal` will be a `Matrix`, which is wrong.
Note only is it the wrong type, but it can contain non-zeros off the diagonal, which is not possible, it escapes the subspace.
```

Reply:
Wait does this one actually have zeros on the off-diagonal? Or is `ȳ ` going to be diagonal?

Maybe a better example is `sum` which has tangent `fill(ȳ , size(A))`

Reply:
No, it is a fully dense matrix in general, since `Diagonal` * `Matrix` is a dense `Matrix`. (Only Diagonal * Diagonal is always a Diagonal)